cmake/GoogleCloudCppLibrary.cmake

```diff
@@ -19,10 +19,21 @@
 # * library:      the short name of the library, e.g. `kms`.
 # * display_name: the display name of the library, e.g. "Cloud Key Management
 #   Service (KMS) API"
+#
+# Additionally, we must set the following **variable** in the parent scope. We
```

Comment:
For a separate PR, maybe we should have a `SERVICE_DIRS` list parameter and a boolean parameter that adds `""` to it, something like `WITH_LOCAL_SERVICE_DIR` or `WITH_FORWARDING_HEADERS`

Reply:
Eh. I think that would add unnecessary complexity. What we have works and I am not trying to get cmake readability.

Reply:
Passing values through global variables in CMake is bad for the same reason that passing values in global variables is bad in all languages.  This is not a good habit to get into.

Reply:
Ack.